x/upgrade/keeper/keeper.go

```diff
@@ -227,6 +233,23 @@ func (k Keeper) GetUpgradedConsensusState(ctx sdk.Context, lastHeight int64) ([]
 	return bz, true
 }
 
+// GetNextOrLastUpgrade returns the last applied upgrade name and height.
+func (k Keeper) GetNextOrLastUpgrade(ctx sdk.Context) (string, int64) {
```

Comment:
it seams we are only returning the last completed upgrade.

Reply:
Yes. renamed to `GetLastCompletedUpgrade`